pkg/storage/unified/search/bleve.go

```diff
@@ -224,6 +231,17 @@ func (b *bleveBackend) cleanOldIndexes(dir string, skip string) {
 	}
 }
 
+// isValidPath does a sanity check in case it tries to access dirs above the file tree
+func isValidPath(path, safeDir string) bool {
+	if strings.Contains(path, "\\") || // if path contains backslashes
+		strings.Contains(path, "..") || // if path is above the safe dir
+		!strings.HasPrefix(path, safeDir) || // if path is not under the safe dir
+		!strings.HasSuffix(path, "/") { // if path is not a directory
+		return false
+	}
+	return true
+}
```

Comment:
This is still exploitable. https://go.dev/play/p/ub2zxsVgFRe -- It's also always good practice to put any edge case that was found during review as part of the tests to validate the future outcome. 

Take a look at [filepath.Clean](https://pkg.go.dev/path/filepath#Clean), that should help you to make your logic more coherent.